src/color.rs

```diff
@@ -69,11 +69,39 @@
 //! use turtle::Color;
 //! let my_color = Color {red: 255.0, green: 55.0, blue: 11.0, alpha: 1.0};
 //! ```
-//!
+//! 
 //! Note that when creating a color this way, we **do not** check if the values of each property are
 //! within their valid ranges.
 //!
-//! A more ergonomic syntax can also be used when passing a color to a method that supports any
+//! There are also constructor methods available for `Color` that allow you to create a new
+//! color using provided values. These are:
+//! 
+//! * `rgb`: Create from the given red, green, and blue values with an alpha value of 1.0
+//! * `rgba`: Similar to `rgb` but also accepts an alpha value
+//! * `hsl`: Create from the given hue, saturation, and lightness values with an alpha of 1.0
+//! * `hsla`: Similar to `hsl` but also accepts an alpha value
+//! 
+//! These methods provide a concise syntax for creating a new `Color` while also providing
+//! validation. All values provided and any color created is checked for correctness.
+//! More information about each can be found on its dedicated documentation.
+//! 
+//! ```rust
+//! use turtle::Color;
+//! 
+//! // These are equivalent
+//! let white_manual = Color { red: 255.0, green: 255.0, blue: 255.0, alpha: 1.0 };
+//! let white_rgb = Color::rgb(255.0, 255.0, 255.0);
+//! let white_rgba = Color::rgba(255.0, 255.0, 255.0, 1.0);
+//! let white_hsl = Color::hsl(0.0, 0.0, 1.0);
+//! let white_hsla = Color::hsla(0.0, 0.0, 1.0, 1.0);
+//! 
+//! assert_eq!(white_manual, white_rgb);
+//! assert_eq!(white_rgb, white_rgba);
+//! assert_eq!(white_rgba, white_hsl);
+//! assert_eq!(white_hsl, white_hsla);
+//! ```
+//! 
+//! Another ergonomic syntax can also be used when passing a color to a method that supports any
```

Comment:
This is awesome! Adding module level documentation is above and beyond and I definitely appreciate you taking the time to do it! :smiley: 

Could you move this entire section of text about the constructors below the other stuff that was there previously? To clarify, I would like you to move all of the docs you just added here to the very end of the documentation that was there previously. Right now, it is above the section about using the colors in the Turtle API. The reason I'm asking for that is because I would like to highlight the usage of the `Color` constructor in Turtle's methods as early as possible. I think everything you've written fits really well if you move it to the very bottom of what is there now.

It may be even better if you change the example to demonstrate the constructors in the context of the Turtle methods rather than just on their own. For example, notice how I use the turtle methods to show how the array syntax for colors is more convenient than the `Color {...}` constructor at times:

```rust
// A solid color with alpha = 1.0
// Syntax is [red, green, blue] and doesn't require explicitly writing the field names
turtle.set_pen_color([133.0, 23.0, 96.0]);
turtle.set_fill_color([133.0, 23.0, 96.0]);
turtle.drawing_mut().set_background_color([133.0, 23.0, 96.0]);
// This is a little easier to type than the equivalent:
turtle.drawing_mut().set_background_color(Color {red: 133.0, green: 23.0, blue: 96.0, alpha: 1.0});
```

If you do that, I would like it if you could mention that "The `Color::rgb` and `Color::rgba` constructors require a little more typing than the array syntax, but might make it a bit easier to remember which color is which argument in case you are having trouble keeping track."

In case you don't recall how it looked before, you can reference the [rendered docs on docs.rs](https://docs.rs/turtle/1.0.0-rc.1/turtle/color/index.html#creating-a-color-from-values)

Reply:
I can, however you originally commented that you wanted the compare/contrast with the literal method of color specification so I wanted to keep them together as it demonstrates the smaller expressive syntax with validation instead of the literal `Color {...}` style

Reply:
We can still do that. Let's just do it at the end of the section rather than pushing the turtle usage down. :smile: